openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java

```diff
@@ -300,6 +300,46 @@ public void shouldCreateProgramEnrollment() {
     }
 
     @Test
+    public void shouldUpdateProgramEnrollment() {
+
+        DateTime dateEnrolled = new DateTime("2000-08-16T07:22:05Z");
+        DateTime dateCompleted = new DateTime("2100-08-16T07:22:05Z");
+
+        ProgramEnrollment.StateStatus state = new ProgramEnrollment.StateStatus();
+        state.setUuid("4b812ac8-421c-470f-b4b7-88187cdbd2a5");
+
+        List<ProgramEnrollment.StateStatus> statuses = new ArrayList<>();
+        statuses.add(state);
+
+        Map<String, String> programEnrollmentAttributes = new HashMap<>();
+        programEnrollmentAttributes.put("8d8718c2-c2cc-11de-8d13-0010c6dffd0f", "testValue");
+
+        List<Attribute> attributes = new ArrayList<>();
+        Attribute attribute = new Attribute();
+        Attribute.AttributeType attributeType = new Attribute.AttributeType();
+
+        attributeType.setUuid("8d8718c2-c2cc-11de-8d13-0010c6dffd0f");
+        attribute.setValue("testValue");
+        attribute.setAttributeType(attributeType);
+
+        attributes.add(attribute);
+
+        ProgramEnrollment programEnrollment = new ProgramEnrollment();
+
+        programEnrollment.setUuid("aaef37f5-ffaa-4f94-af6e-54907b0c0fd4");
+        programEnrollment.setDateEnrolled(dateEnrolled.toDate());
+        programEnrollment.setDateCompleted(dateCompleted.toDate());
+        programEnrollment.setStates(statuses);
+        programEnrollment.setAttributes(attributes);
+
+        openMRSActionProxyService.updateProgramEnrollment(CONFIG_NAME, programEnrollment.getUuid(), dateCompleted,
+                state.getUuid(), dateEnrolled, programEnrollmentAttributes);
+
+        verify(programEnrollmentService).updateProgramEnrollment(eq(CONFIG_NAME), programEnrollmentCaptor.capture());
+        assertEquals(programEnrollment, programEnrollmentCaptor.getValue());
```

Comment:
ProgramEnrollment does not have attributes in equals method, so this check does not verify attributes